### PR TITLE
feat(ci): introduce slow e2e test step

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As much as possible, we'd like our test suite to be fast. But some tests are slow due to their testing properties, for instance timeout and rate limit.

As such we partition these tests into a separate step to parallelize their execution better.

I will migrate backfill tests to this step as well later.

Related: https://github.com/risingwavelabs/risingwave/issues/16897

Closes https://github.com/risingwavelabs/risingwave/issues/16940

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by kwannoel** [comment]
Bump

**Review by lmatz — APPROVED** [review]
LGTM

**Comment by lmatz** [comment]
does `release-1.9` require this?

**Comment by kwannoel** [comment]
> does `release-1.9` require this?

I don't think it's a hard req

File: ci/scripts/run-e2e-test.sh [hunk 1]
```diff
@@ -115,7 +114,6 @@ pkill python3
 
 sqllogictest -p 4566 -d dev './e2e_test/udf/alter_function.slt'
 sqllogictest -p 4566 -d dev './e2e_test/udf/graceful_shutdown_python.slt'
-sqllogictest -p 4566 -d dev './e2e_test/udf/always_retry_python.slt'
```
Comment (xxchan) [inline ci/scripts/run-e2e-test.sh:118]: There's one concern: We cannot distinguish which tests are run in which pipeline. Ideally we should separate them into different directories like `e2e_test_slow`.

**Comment by lmatz** [comment]
https://buildkite.com/risingwavelabs/main-cron/builds/2545#018fbedc-b24e-462d-8c71-7800c4b9153e
SCR-20240528-stj

let me label it

File: ci/scripts/run-e2e-test.sh (hunk 1 above)
Comment (kwannoel) [reply ci/scripts/run-e2e-test.sh:118]: Sure, I think we could do that.

Comment (kwannoel) [reply ci/scripts/run-e2e-test.sh:118]: Done

Comment (kwannoel) [reply ci/scripts/run-e2e-test.sh:118]: 09b94e50de22674da69cf47bed0ec8c4ad3b5bd1

File: ci/scripts/slow-e2e-test.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+echo "--- Download artifacts"
+mkdir -p e2e_test/udf/wasm/target/wasm32-wasi/release/
+buildkite-agent artifact download udf.wasm e2e_test/udf/wasm/target/wasm32-wasi/release/
+buildkite-agent artifact download udf.jar ./
+
+echo "--- e2e, $mode, slow-udf-tests"
+python3 -m pip install --break-system-packages arrow-udf==0.2.1
+RUST_LOG="info" \
+risedev ci-start "$mode"
```
Comment (xxchan) [inline ci/scripts/slow-e2e-test.sh:38]: Why do we want to add this config, which is different from the conventions of other steps?

**Review by xxchan — APPROVED** [review]
Generally LGTM, thanks!

File: ci/scripts/slow-e2e-test.sh (hunk 2 above)
Comment (kwannoel) [reply ci/scripts/slow-e2e-test.sh:38]: Its similar to what we do in e2e-test, so we can customize the cluster type we are running, inside the workflow. E.g. `ci-3streaming-2serving-3fe`.

**Pull request merged**